src/NuGet.Core/NuGet.Commands/CommandRunners/PackCommandRunner.cs

```diff
@@ -506,24 +507,38 @@ private static void LoadProjectJsonFile(PackageBuilder builder, string path, str
 
         private static void CalculateExcludes(IncludeExcludeFiles files, out string fullExclude, out string filesExclude)
         {
-            fullExclude = string.Empty;
+            var fullExcludeBuilder = new StringBuilder();
             filesExclude = string.Empty;
             if (files.Exclude != null &&
                 files.Exclude.Any())
             {
-                fullExclude = string.Join(";", files.Exclude);
+                bool first = true;
+                foreach (var exclude in files.Exclude)
+                {
+                    if (first)
+                    {
+                        first = false;
+                    }
+                    else
+                    {
+                        fullExcludeBuilder.Append(';');
+                    }
+                    fullExcludeBuilder.Append(exclude);
+                }
```

Comment:
String joining is a common pattern, and I'd hate to have to repeat this `foreach` loop all over the place.

How about creating in NuGet.Common a `StringBuilder` extension that matches [one of these signatures](https://docs.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.appendjoin?view=netstandard-2.1), so that eventually we can remove the extension method and get the behavior directly from the framework?

Be sure to add tests for the extension method.